src/python/pants/backend/native/rules/targets.py

```diff
@@ -0,0 +1,230 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import Dict, Iterable, Optional, Tuple, Union
+
+from pants.backend.native.subsystems.native_build_step import ToolchainVariant
+from pants.backend.native.targets.external_native_library import ConanRequirement
+from pants.backend.native.targets.native_artifact import NativeArtifact
+from pants.build_graph.address import Address
+from pants.engine.target import (
+    COMMON_TARGET_FIELDS,
+    BoolField,
+    Dependencies,
+    InvalidFieldException,
+    InvalidFieldTypeException,
+    PrimitiveField,
+    Sources,
+    StringField,
+    StringSequenceField,
+    Target,
+)
+from pants.util.collections import ensure_list, ensure_str_list
+from pants.util.frozendict import FrozenDict
+
+
+class CSources(Sources):
+    default = ("*.h", "*.c")
+
+
+class CppSources(Sources):
+    default = ("*.h", "*.hpp", "*.cpp")
+
+
+class CtypesNativeLibrary(PrimitiveField):
+    alias = "ctypes_native_library"
+    value: NativeArtifact
+    required = True
+
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[NativeArtifact], *, address: Address
+    ) -> NativeArtifact:
+        value = super().compute_value(raw_value, address=address)
+        if not isinstance(value, NativeArtifact):
+            raise InvalidFieldTypeException(
+                address, cls.alias, value, expected_type="a `native_artifact` object"
+            )
+        return value
+
+
+class FatalWarnings(BoolField):
```

Comment:
@cosmicexplorer do you have any docstring I could add to this so that it shows up in the output of `target-types2`? No worries if not.